### PR TITLE
gitlab builds for osx and win

### LINKED ISSUE #98: Set up binaries signing for windows/osx
Probably should sign our binaries on mac and windows. Not sure about linux?

### PR DESCRIPTION
**Pull request by sachaaaaa**
Get gitlab build binaries and run tests on osx and windows.
On windows, all the dependencies have been installed on the build machine, under the Administrator user, to speed up the job execution time.
On mac, only node/npm has been installed.

There might be some spurious errors in the test on osx, needs investigating.
Fixed oxen-io/session-desktop-temp#1219.

### CODE REVIEW

**Comment by Mikunj** [comment]
```
There might be some spurious errors in the test on osx, needs investigating.
```
Could you give an example?

**Comment by sachaaaaa** [comment]
Yeah:
https://gitlab.com/lokiproject/loki-messenger/-/jobs/133884318
This one failed due to errors:
```
[27478:1213/155256.713122:INFO:CONSOLE(112)] "ERROR", source: /Users/CDCI/Builds/builds/113db301/0/lokiproject/loki-messenger/js/logging.js (112)
{"name":"log","hostname":"Lokis-Mac-mini.local","pid":27478,"level":50,"msg":"Top-level unhandled error: TypeError: describe(...) is not a function\n    at file://[REDACTED]/loki-messenger/test/models/conversations_test.js:195:3","time":"2018-12-13T04:52:56.713Z","v":0}
```

https://gitlab.com/lokiproject/loki-messenger/-/jobs/133888721
This one just got stuck at  `Running "unit-tests" task`, I manually cancelled it after 24 minutes, maybe should've waited longer?

**Comment by Mikunj** [comment]
Right then do we want to disable tests in gitlab and use it exclusively for builds since we have travis running the tests?

**Comment by sachaaaaa** [comment]
ok probably easier yeah.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,25 +1,61 @@
...
+    # - type package.json | findstr /v certificateSubjectName > temp.json
+    # - move temp.json package.json
+    - call  yarn prepare-beta-build
+    - call node_modules\.bin\build --config.extraMetadata.environment=%SIGNAL_ENV% --publish=never
+  cache:
+    paths:
+      - node_modules/
+  artifacts:
+    paths:
+      - dist/
```
Comment (BeaudanBrown) [inline .gitlab-ci.yml]:
```suggestion
      - dist/

```

**Review by BeaudanBrown — APPROVED** [review]

**Pull request merged**